src/stream/src/executor/join/hash_join.rs

```diff
@@ -1071,10 +1079,10 @@ impl<E: JoinEncoding> JoinEntryState<E> {
             }
         } else {
             let mut pk_set = JoinRowSet::default();
-            pk_set.try_insert(pk, ()).unwrap();
+            pk_set.try_insert(pk, ()).expect("pk set should be empty");
             self.inequality_index
                 .try_insert(inequality_key, pk_set)
-                .unwrap();
+                .expect("pk set should be empty");
```

Comment:
Don't want to change the signature, it's used by ASOF Join. Just add an expect here.